### PR TITLE
added control for the arrayRenderer, consolidate module, added styles

### PR DESCRIPTION
**Pull request by rsoika**
also removed the file ArrayControl tsx and consolidated in ArrayControlRenderer.tsx

### CODE REVIEW

**Comment by rsoika** [comment]
This is the new layout example:

image

**Comment by rsoika** [comment]
Hi @sdirix  I don't want to be impatient, but what did you think about my last pull request?

**Comment by sdirix** [comment]
Hi @rsoika, we'll take a look :+1:

File: packages/vanilla/src/styles/styles.ts [hunk 1]
```diff
@@ -145,10 +145,30 @@ export const vanillaStyles: StyleDef[] = [
...
+  },  
+  {
+    name: 'array.child.controls.delete',
+    classNames: ['button-delete']
+  },  
   {
     name: 'array.control',
     classNames: ['array-control-layout', 'control']
-  },  
+  },
```
Comment (lucas-koehler) [inline packages/vanilla/src/styles/styles.ts:171]: Please remove the whitespace following the comma

File: packages/vanilla/src/complex/array/index.ts [hunk 2]
```diff
@@ -28,9 +28,8 @@ import {
   rankWith
 } from '@jsonforms/core';
 import ArrayControlRenderer from './ArrayControlRenderer';
-import { ArrayControl } from './ArrayControl';
 
-export { ArrayControl, ArrayControlRenderer };
+export {  ArrayControlRenderer };
```
Comment (lucas-koehler) [inline packages/vanilla/src/complex/array/index.ts:32]: Similarily to the comment in `ArrayControlRenderer.tsx`, I think we should export `ArrayControl` here.
You can get import it together with the `ArrayControlRenderer` like this:
`import ArrayControlRenderer, { ArrayControl } from './ArrayControlRenderer';`
and the change the export to
`export {  ArrayControlRenderer, ArrayControl };`

File: packages/vanilla/src/complex/array/ArrayControlRenderer.tsx [hunk 3]
```diff
@@ -22,19 +22,114 @@
...
+      </header>
+      <div className={divClassNames}>
+        {errors}
+      </div>      
+      <div className={classNames.children}>
+        {data ? (
+          range(0, data.length).map(index => {
+            const childPath = composePaths(path, `${index}`);
+            return (
+              <div key={index}><JsonFormsDispatch
```
Comment (lucas-koehler) [inline packages/vanilla/src/complex/array/ArrayControlRenderer.tsx:91]: Please add a new line before `<JsonFormsDispatch` to increase readability

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
Hi @rsoika ,
thank you very much for your pull request. In general, this already works great. I tested the changes with the vanilla dev example and did not run into any issues.

I added some comments regarding formatting and I think we should still export the `ArrayControl` to allow adopters to use it in their custom renderers and to not break the API in this regard.

**Comment by rsoika** [comment]
Hi @lucas-koehler, I have fixed now the issues you mentioned.

**Review by lucas-koehler — APPROVED** [review]
@rsoika Thank you for the updates and thanks again for your contribution. LGTM now :)

**Pull request merged**